src/imageclipper.cpp

```diff
@@ -43,835 +44,745 @@
-                                    cvRect32fFromRect( param->rect, param->rotate ), 
-                                    cvPointTo32f( param->shear ) );
-            }
-        }
-    }
+	cout << "ImageClipper - image clipping helper tool." << endl;
+	cout << "Command Usage: " << fs::basename(arg->name);
+	cout << " [option]... [arg_reference]" << endl;
+	cout << "  <arg_reference = " << arg->reference << ">" << endl;
+	cout << "    <arg_reference> would be a directory or an image or a video filename." << endl;
+	cout << "    For a directory, image files in the directory will be read sequentially." << endl;
+	cout << "    For an image, it starts to read a directory from the specified image file. " << endl;
+	cout << "    (A file is judged as an image based on its filename extension.)" << endl;
+	cout << "    A file except images is tried to be read as a video and read frame by frame. " << endl;
+	cout << endl;
+	cout << "  Options" << endl;
+	cout << "    -o <output_format = imgout_format or vidout_format>" << endl;
+	cout << "        Determine the output file path format." << endl;
+	cout << "        This is a syntax sugar for -i and -v. " << endl;
+	cout << "        Format Expression)" << endl;
+	cout << "            %d - dirname of the original" << endl;
+	cout << "            %i - filename of the original without extension" << endl;
+	cout << "            %e - filename extension of the original" << endl;
+	cout << "            %x - upper-left x coord" << endl;
+	cout << "            %y - upper-left y coord" << endl;
+	cout << "            %w - width" << endl;
+	cout << "            %h - height" << endl;
+	cout << "            %r - rotation degree" << endl;
+	cout << "            %. - shear deformation in x coord" << endl;
+	cout << "            %, - shear deformation in y coord" << endl;
+	cout << "            %f - frame number (for video)" << endl;
+	cout << "        Example) ./$i_%04x_%04y_%04w_%04h.%e" << endl;
+	cout << "            Store into software directory and use image type of the original." << endl;
+	cout << "    -i <imgout_format = " << arg->imgout_format << ">" << endl;
+	cout << "        Determine the output file path format for image inputs." << endl;
+	cout << "    -v <vidout_format = " << arg->vidout_format << ">" << endl;
+	cout << "        Determine the output file path format for a video input." << endl;
+	cout << "    -f" << endl;
+	cout << "    --frame <frame = 1> (video)" << endl;
+	cout << "        Determine the frame number of video to start to read." << endl;
+	cout << "    -h" << endl;
+	cout << "    --help" << endl;
+	cout << "        Show this help" << endl;
+	cout << endl;
+	cout << "  Supported Image Types" << endl;
+	cout << "      bmp|dib|jpeg|jpg|jpe|png|pbm|pgm|ppm|sr|ras|tiff|exr|jp2" << endl;
 }
 
 /**
-* cvSetMouseCallback function
+* Print Application Usage
 */
-void mouse_callback( int event, int x, int y, int flags, void* _param )
+void printUsage()

```

Comment:
I'd prefer if you did not rearrange the functions like this for no reason. It simply makes a messier diff. Please move them back... If the reason is that you need to have a function declared before you use it in another function, then do that in a header.

I might want to backport some old patches, and changes like this makes doing that really hard